src/framework/auth/services/token/token.ts

```diff
@@ -80,6 +97,20 @@ export class NbAuthJWTToken extends NbAuthSimpleToken {
 
   static NAME = 'nb:auth:jwt:token';
 
+  /**
+   * for JWT token, the iat (issued at) field of the token payload contains the creation Date
+   */
+  protected prepareCreatedAt(date: Date) {
+    let decoded = null;
+    try { // needed as getPayload() throws error and we want the token to be created in any case
+      decoded = this.getPayload();
+    }
+    finally {
+      return decoded && decoded.iat ? new Date(Number(decoded.iat) * 1000) : (date ? date : new Date());
+
```

Comment:
unnecessary new line
I would suggest the following change
 
```
protected prepareCreatedAt(date: Date) {
  date = super.prepareCreatedAt(date); // get or otherwise create
  let decoded = null;
  try { // needed as getPayload() throws error and we want the token to be created in any case
    decoded = this.getPayload();
  } finally {
    return decoded && decoded.iat ? new Date(Number(decoded.iat) * 1000) : date;
  }
}
```